tensorly/metrics/similarity.py

```diff
@@ -0,0 +1,81 @@
+from typing import List
+
+import numpy as np
+from sklearn.preprocessing import normalize
+
+# Authors: Hratch Baghdassarian <hmbaghdassarian@gmail.com>, Erick Armingol <earmingol14@gmail.com>
+# similarity metrics for tensor decompsoitions
+
+def stack_loading_matrices(factors: List[np.ndarray]) -> np.ndarray:
+    """Vertically stack the loading/factor matrices.
+
+    Parameters
+    ----------
+    factors : List[np.ndarray]
+        The loading/factor matrices [A^1 ... A^R] for a decomposed low-rank tensor from its factors.
+
+    Returns
+    -------
+    np.ndarray
+        A vertically stacked matrix of shape [sum(tensor.shape)xR)] 
+    """
+    
+    if not np.all([A.shape[1] for A in factors]):
+        raise ValueError('factors should be a list of loading matrices')
+    return np.vstack(factors)
+
+def calc_corrindex(C: np.ndarray) -> float:
+    """Calculate the CorrIndex.
+
+    Parameters
+    ----------
+    C : np.ndarray
+        absolute value of the product of two MxN matrices that have had their columns L2 normalized, 
+        taking the complex conjugate transpose of the first matrix -- C = |X1^H * X2|
+
+    Returns
+    -------
+    float
+        CorrIndex metric [0,1]; lower score means higher between matrices
+    """
+    
+    MN = C.shape[1]*C.shape[0]
```

Comment:
Given that C is generated from C = np.abs(np.matmul(np.matrix(X_1).H, X_2)) (in Line 76) and both X_1, X_2 must be of shape M x N,  C should be of shape N x N. So C.shape[1] should be equal to C.shape[0]. Maybe include an assert condition before this.